File: Data Structures & Algorithms/search-2d-matrix/submission-1.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n=matrix.size(),m=matrix[0].size();

            int s=0,e=n*m-1;
            while(s<=e){
                int mid=(s+e)/2;
                int r=mid/m,c=mid%m;

                if(matrix[r][c]<target)s=mid+1;
                else if (matrix[r][c]>target)e=mid-1;
                else return true;
          }
return false;
        // int r=0,c=m-1;
        // while(r<n and c>=0){
        //     if(matrix[r][c]>target){
        //         c--;
        //     }
        //     else if(matrix[r][c]==target)return true;
        //     else r++;
        // }   
        // return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:

    def bs(self, a : List[int] , target :int ,start: int )->int:
        s=start 
        e=len(a)-1
        while s<=e:
            m=s+(e-s)//2;
            if a[m]==target:
                return m
            elif a[m]<target:
                s=m+1
            else:
                e=m-1
        return -1            
    
    
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i=0
        j=len(numbers)-1
        ans=[]
        while i<=j:
            x=target-numbers[i]
            a=self.bs(numbers ,x,i+1)
            if a!=-1:
                return [i+1,a+1]

            i+=1

        return []